home_assistant_streamdeck_yaml.py

```diff
@@ -2253,34 +2344,136 @@ async def key_change_callback(
         console.log(f"Key {key} {'pressed' if key_pressed else 'released'}")
 
         button = config.button(key)
-        assert button is not None
-        if button is not None and key_pressed:
+        if button is None:
+            console.log(f"No button found for key {key}")
+            return
 
-            async def cb() -> None:
-                """Update the deck once more after the timer is over."""
-                assert button is not None  # for mypy
-                await _handle_key_press(websocket, complete_state, config, button, deck)
+        if key_pressed:
+            press_start_times[key] = time.time()
+            console.log(
+                f"Key {key} pressed, starting long press monitor with threshold {long_press_threshold}s",
+            )
+            update_key_image(
+                deck,
+                key=key,
+                config=config,
+                complete_state=complete_state,
+                key_pressed=True,
+            )
+            coro = _monitor_long_press(
+                key=key,
+                press_start_times=press_start_times,
+                long_press_threshold=long_press_threshold,
+                websocket=websocket,
+                complete_state=complete_state,
+                config=config,
+                button=button,
+                deck=deck,
+            )
+            press_tasks[key] = asyncio.create_task(coro)
+            return
 
-            if button.maybe_start_or_cancel_timer(cb):
-                key_pressed = False  # do not click now
```

Comment:
Note to self: check if this behavior is still there.

Reply:
@ngolf previously (see https://github.com/basnijholt/home-assistant-streamdeck-yaml/blob/3d793f42422fb71492e19cdac0c1074a27369691/home_assistant_streamdeck_yaml.py#L2240-L2249)

When `button.delay is True`, the `update_key_image` function was called with `key_pressed=False`, now the behavior is changed. I have to admit I do not remember (it's been a few years) if that was intentional, so you _**might**_ have changed it to the correct thing.

Since I am not at home with my StreamDeck I have no way of testing that.

What do you think about it?